### PR TITLE
diagnostic_msgs: Add services for dynamic addition, removal and listing of analyzers

### PR DESCRIPTION
**Pull request by heuristicus**
See ros/diagnostics/pull/41 for the code which uses these new srvs.

### CODE REVIEW

File: diagnostic_msgs/srv/AddDiagnostics.srv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+string namespc
+---
+bool success
```
Comment (jack-oquin) [inline diagnostic_msgs/srv/AddDiagnostics.srv]: Missing newline

File: diagnostic_msgs/srv/ListDiagnostics.srv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+string[] names
```
Comment (jack-oquin) [inline diagnostic_msgs/srv/ListDiagnostics.srv]: Missing newline

File: diagnostic_msgs/srv/RemoveDiagnostics.srv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+string namespc
+---
+bool success
```
Comment (jack-oquin) [inline diagnostic_msgs/srv/RemoveDiagnostics.srv]: Missing newline

File: diagnostic_msgs/CMakeLists.txt [hunk 4]
```diff
@@ -13,7 +13,10 @@ add_message_files(
 add_service_files(
   DIRECTORY srv
   FILES
-  SelfTest.srv)
+  SelfTest.srv
+  AddDiagnostics.srv
+  RemoveDiagnostics.srv
+  ListDiagnostics.srv)
```
Comment (jack-oquin) [inline diagnostic_msgs/CMakeLists.txt]: Please alphabetize the names.

**Comment by jack-oquin** [comment]
This sounds like a good idea, but deserves discussion on the ros-sig-drivers mailing list. A summary and links to the relevant issues should suffice.

**Comment by stonier** [comment]
@heuristicus please close this one...#80 is the currently relevant one.

**Pull request closed without merging**